src/libraries/System.Private.CoreLib/src/System/IO/FileSystemInfo.cs

```diff
@@ -107,6 +115,46 @@ public DateTime LastWriteTimeUtc
             set => LastWriteTimeCore = File.GetUtcDateTimeOffset(value);
         }
 
+        /// <summary>
+        /// If this <see cref="FileSystemInfo"/> instance represents a link, returns the link target's path.
+        /// If a link does not exist in <see cref="FullName"/>, or this instance does not represent a link, returns <see langword="null"/>.
+        /// </summary>
+        public string? LinkTarget => _linkTarget ??= FileSystem.GetLinkTarget(FullPath, this is DirectoryInfo);
+
+        /// <summary>
+        /// Creates a symbolic link located in <see cref="FullName"/> that points to the specified <paramref name="pathToTarget"/>.
+        /// </summary>
+        /// <param name="pathToTarget">The path of the symbolic link target.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="pathToTarget"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="pathToTarget"/> is empty.
+        /// -or-
+        /// This instance was not created passing an absolute path.
+        /// -or-
+        /// <paramref name="pathToTarget"/> contains invalid path characters.</exception>
+        /// <exception cref="IOException">A file or directory already exists in the location of <see cref="FullName"/>.
+        /// -or-
+        /// An I/O error occurred.</exception>
+        public void CreateAsSymbolicLink(string pathToTarget)
+        {
+            FileSystem.VerifyValidPath(pathToTarget, nameof(pathToTarget));
+            FileSystem.CreateSymbolicLink(OriginalPath, pathToTarget, this is DirectoryInfo);
+            Invalidate();
+        }
+
+        /// <summary>
+        /// Gets the target of the specified link.
+        /// </summary>
+        /// <param name="returnFinalTarget"><see langword="true"/> to follow links to the final target; <see langword="false"/> to return the immediate next link.</param>
+        /// <returns>A <see cref="FileSystemInfo"/> instance if the link exists, independently if the target exists or not; <see langword="null"/> if this file or directory is not a link.</returns>
+        /// <exception cref="IOException">The file or directory does not exist.
+        /// -or-
+        /// The link's file system entry type is inconsistent with that of its target.
+        /// -or-
+        /// Too many levels of symbolic links.</exception>
+        /// <remarks>When <paramref name="returnFinalTarget"/> is <see langword="true"/>, the maximum number of symbolic links that are followed are 40 on Unix.</remarks>
+        public FileSystemInfo? ResolveLinkTarget(bool returnFinalTarget = false) =>
```

Comment:
@adamsitnik as mentioned in https://github.com/dotnet/runtime/issues/24271#issuecomment-851635211, I think the appropriate default is `true` here.

Imo it's more interesting to figure out the canonical path than what this method will give you. This was proposed as a method on `Path` in https://github.com/dotnet/runtime/issues/24271#issuecomment-849065336 and refined by @mklement0 in https://github.com/dotnet/runtime/issues/24271#issuecomment-670709364.

Reply:
>  I think the appropriate default is true here

I agree. @Jozkee should we just change it before it gets merged so we don't need to introduce breaking changes in the future?